tests/integration/tables/windows_firewall_rules.cpp

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+// Sanity check integration test for connectivity
+// Spec file: specs/windows/windows_firewall_rules.table
+
+#include <osquery/tests/integration/tables/helper.h>
+
+namespace osquery {
+namespace table_tests {
+
+class windows_firewall_rules : public testing::Test {
+ protected:
+  void SetUp() override {
+    setUpEnvironment();
+  }
+};
+
+TEST_F(windows_firewall_rules, test_sanity) {
+  auto const data =
+      execute_query("select * from windows_firewall_rules LIMIT 1");
+
+  ASSERT_EQ(data.size(), 1ul);
+
+  ValidationMap row_map = {
```

Comment:
Once the code has been refactored, a unit test with mocked input data should make sure that the rendering works as expected with different input values